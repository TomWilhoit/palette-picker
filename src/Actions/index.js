export const addProjects = (projects) => ({
  type: 'ADD_PROJECTS',
  projects
})

export const addPalettes = (palettes) => ({
  type: 'ADD_PALETTES',
  palettes
})

export const addCurrentProject = (project) => ({
  type: 'ADD_CURRENT_PROJECT',
  project
})