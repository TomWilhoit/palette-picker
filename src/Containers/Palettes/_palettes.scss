.palettes-container {
  @include container;
  background-color: $light-background;
  display: block;
  height: 96%;
  margin-left: 10%;
  overflow-y: scroll;
  width: 80%;
}