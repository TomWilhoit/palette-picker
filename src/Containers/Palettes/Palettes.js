import React, { Component } from "react";
import Palette from "../Palette/Palette";
import { connect } from "react-redux";
import { updateCurrentPalette } from "../../Actions";
import PropTypes from "prop-types";

export class Palettes extends Component {

  renderPalettes = currProject => {
    if (this.props.palettes.length) {
      const palettes = this.props.palettes.filter(palette => palette.project_id === currProject);
      return palettes.map(palette => (
        <Palette
          color1={palette.color1}
          color2={palette.color2}
          color3={palette.color3}
          color4={palette.color4}
          color5={palette.color5}
          id={palette.id}
          key={palette.id}
          name={palette.name}
          setError={this.props.setError} 
          setPaletteDisplay={this.props.setPaletteDisplay}
          showPaletteName={this.props.showPaletteName}
        />
      ));
    }
  }

  render() {
    return (
      <div className="palettes-console">
        <div className="create-new-palette">
          <Palette
            color1="FEFEFE"
            color2="FEFEFE"
            color3="FEFEFE"
            color4="FEFEFE"
            color5="FEFEFE" 
            id={0}
            key={0}
            name="Create New Palette"
            setError={this.props.setError} 
            setPaletteDisplay={this.props.setPaletteDisplay}
            showPaletteName={this.props.showPaletteName}
          />
        </div>
        <div className="saved-palettes scroll">
          {!this.props.palettes.length &&
            <div className="no-palettes">
              <p>No saved palettes.</p>
              <p>Select a project <i className="fas fa-arrow-up" /></p>
              <p>Mix/Submit palette <i className="fas fa-arrow-down" /></p>
            </div>
          }
          {this.renderPalettes(this.props.currentProject)}
        </div>
      </div>
    );
  }
}

Palettes.propTypes = {
  setPaletteDisplay: PropTypes.func,
  setError: PropTypes.func,
};

export const mapStateToProps = state => ({
  currentProject: state.currentProject,
  palettes: state.palettes,
});

export const mapDispatchToProps = dispatch => ({
  updateCurrentPalette: palette => dispatch(updateCurrentPalette(palette)),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Palettes);