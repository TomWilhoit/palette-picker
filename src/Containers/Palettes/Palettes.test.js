import React from "react";
import { Palettes } from "./Palettes";
import { mapDispatchToProps, mapStateToProps } from "./Palettes";
import ReactDOM from "react-dom";
import { shallow } from "enzyme";
import { updateCurrentPalette } from "../../Actions";

describe("Palettes", () => {
  let wrapper;
  let mockPalettes = [{ name: "Tom" ,projectId: 4}];
  
  beforeEach(() => {
    wrapper = shallow(<Palettes palettes={mockPalettes} />);
  });

  it("should match the snapshot with all data passed in", () => {
    expect(wrapper).toMatchSnapshot();
  })

  
  it("should mapStateToProps", () => {
    const mockState = {
      currentProject: 4,
      palettes: [{ name: "Mason", projectId: 4 }],
      currentProject: 4
    };
    const mappedProps = mapStateToProps(mockState);
    expect(mappedProps).toEqual(mockState);
  })

  it("should map dispatch to props", () => {
    const mockPalette= {name: "Tommy", projectId: 4};
    const mockDispatch = jest.fn();
    const actionToDispatch = updateCurrentPalette(mockPalette);
    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.updateCurrentPalette(mockPalette);
    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
  })
})