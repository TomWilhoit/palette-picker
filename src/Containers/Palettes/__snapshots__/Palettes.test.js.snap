// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Palettes should match the snapshot with all data passed in 1`] = `
<div
  className="palettes-container"
>
  <ConnectFunction
    id={0}
    key="0"
    name="New Palette"
    refreshSelectedPalette={[Function]}
  />
  <ConnectFunction
    name="Tom"
  />
</div>
`;
